modules/opengl/src/shader/shaderobject.cpp

```diff
@@ -373,31 +373,41 @@ void ShaderObject::compile() {
 #endif
 }
 
-void ShaderObject::addShaderDefine(std::string name, std::string value) {
+void ShaderObject::addShaderDefine(const std::string& name, const std::string& value) {
     shaderDefines_[name] = value;
 }
+void ShaderObject::setShaderDefine(const std::string& name, bool exists, const std::string& value) {
+    if (exists) {
```

Comment:
addOrRemoveshaderDefine has the right meaning... but it looks so ugly ;) How about toggleShaderDefine?

Reply:
Sure, sounds better. Without any bool then :)

Reply:
No I need the bool, I want to do this:

    shader[ShaderType::Vertex]->setShaderDefine("UNIFORM_RADIUS", overrideSphereRadius_);

instead of this:

    if (overrideSphereRadius_.get()) {
        shader_.getVertexShaderObject()->addShaderDefine("UNIFORM_RADIUS");
    } else {
        shader_.getVertexShaderObject()->removeShaderDefine("UNIFORM_RADIUS");
    }

We have this pattern in quite many places and it is very verbose...

Reply:
ok, but toggleDefine(define, false) does not make sense either. Then setShaderDefine is better even if it does not do what it says :) 
Perhaps there is another option? Will think about it

Reply:
I was considering configShaderDefine or updateShaderDefine